Comparison of integers of different signs

When compiling NJOY21, I get the following compilation error:

```console
/Users/jlconlin/NJOY21/Code/NJOY21/bin-llvm/_deps/endftk-src/src/ENDFtk/section/1/455/DecayConstants/src/generateList.hpp:11:12: error: comparison of
      integers of different signs: 'int' and 'std::__1::vector<double, std::__1::allocator<double> >::size_type' (aka 'unsigned long')
      [-Werror,-Wsign-compare]
  if ( nnf != abundances.size() ) {
       ~~~ ^  ~~~~~~~~~~~~~~~~~
/Users/jlconlin/NJOY21/Code/NJOY21/bin-llvm/_deps/endftk-src/src/ENDFtk/section/1/455/DecayConstants/src/generateList.hpp:21:31: error: comparison of
      integers of different signs: 'unsigned int' and 'int' [-Werror,-Wsign-compare]
  for ( unsigned int i = 0; i < nnf; ++i ) {
                            ~ ^ ~~~
```

I'm not sure why this doesn't show up when compiling the tests.

--- a/src/ENDFtk/section/1/455/DecayConstants/src/generateList.hpp
+++ b/src/ENDFtk/section/1/455/DecayConstants/src/generateList.hpp
@@ -2,7 +2,7 @@ static std::vector< double >
 generateList( std::vector< double >&& constants,
               std::vector< double >&& abundances ) {
 
-  int nnf = constants.size();
+  auto nnf = constants.size();
   if ( nnf == 0 ) {
 
     Log::error( "The number of precursor families NNF cannot be zero" );

